ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh

```diff
@@ -70,3 +70,39 @@ EOF
 
 ansible-playbook packet-setup.yaml -e "packet_hostname=ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"
 
+cat << EOF | tee "${SHARED_DIR}/packet-conf"
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [ -x "\$(command -v nss_wrapper.pl)" ]; then
+        grep -v -e ^default -e ^\$(id -u) /etc/passwd > "/tmp/passwd"
+        echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/tmp/passwd"
+        export LD_PRELOAD=libnss_wrapper.so
+        export NSS_WRAPPER_PASSWD=/tmp/passwd
+        export NSS_WRAPPER_GROUP=/etc/group
+    elif [[ -w /etc/passwd ]]; then
+        echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/etc/passwd"
+    else
+        echo "No nss wrapper, /etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+# Initial check
+if [ "\${CLUSTER_TYPE}" != "packet" ]; then
+    echo >&2 "Unsupported cluster type '\${CLUSTER_TYPE}'"
+    exit 1
+fi
+
+IP=\$(cat "\${SHARED_DIR}/server-ip")
+SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -i "\${CLUSTER_PROFILE_DIR}/.packet-kni-ssh-privatekey")
+
+# Checkout packet server
+for x in \$(seq 10) ; do
+    test "\${x}" -eq 10 && exit 1
+    ssh "\${SSHOPTS[@]}" "root@\${IP}" hostname && break
+    sleep 10
+done
+
+EOF
```

Comment:
Even though this approach puts in common some points (initial check, ip extraction and wait for packet ready), it seems quite an overhead in respect to the rest to the other setup/gather steps. I think it could be more useful (in terms of maintenance) moving into a direction of having just one version (rather than three flavors as for now) - and this would probably mean addressing the main issue related to the fact that the assisted/sno steps are currently using a different image

Reply:
I tried to make it common at first, by only converging the nvme mount. But each project has other prerequisites.
For example, dev-scripts is expected to be set in `/opt/dev-scripts`. Although it can be overridden with the [WORKING_DIR](https://github.com/openshift-metal3/dev-scripts/blob/cc9e6d591ad21f39848852899fd130093933ec12/common.sh#L72) environment variable, this is just an example of how a repository expects stuff that the other isn't (assisted can be running from whenever it is being deployed and it doesn't utilize a `WORKDIR_DIR` env var).

I think a better approach instead of changing other repositories to a certain methodology, just in order to use packet - Packet setup step should just be easily useable.
Calling packet-setup & packet-gather are mandatory.
IMO instead of a lot of verifications, assignments, loops, etc on the following step - simple `source {SHARED_DIR}/packet-conf` seems really easy to be used. After that, anyone would be able to use that packet server.

Reply:
I think nvme mount could be a good point. Probably there are some other (small) points that we can work out to leverage the differences - easier than the image used, and I'd personally really like to move into that direction. Given that, I think that this first step could be fine in general, even though in the long term we'll need something else (but I guess we're on sync on that).